python/taichi/lang/matrix.py

```diff
@@ -1793,7 +1793,6 @@ def __init__(self, n, dtype, shape, layout):
         self.layout = layout
         self.shape = tuple(shape)
         self.element_type = TensorType((n, ), self.dtype)
-        # TODO: pass in element_type, shape, layout directly
```

Comment:
This TODO still applies IMHO : in the ideal world we'll want to pass in the whole element_type  once we have vec3/matrix types in C++. But now we'll have to split it as  dtype + shape :P